### PR TITLE
New package: deno-1.5.3

### PR DESCRIPTION
**Pull request by shizonic**

### CODE REVIEW

File: srcpkgs/deno/template [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Template file for 'deno'
+pkgname=deno
+version=1.4.4
+revision=1
+build_style=cargo
+makedepends="python ccache"
```
Comment (Chocimier) [inline srcpkgs/deno/template:6]: ccache is not appropriate for CI nor official builders. For local builds, configure xbps-src in etc/conf as described in README.md.

**Comment by Chocimier** [comment]
Are problems from #22043 addressed in this release?

**Comment by shizonic** [comment]
No they aren't. I can't figure out where exactly the binaries are fetched.

File: srcpkgs/deno/template (hunk 1 above)
Comment (shizonic) [reply srcpkgs/deno/template:6]: Gonna change this.

**Comment by shizonic** [comment]
I really don't know why it always runs out of time. It builds successfully on my local machine. Any suggestions?

**Comment by Chocimier** [comment]
Disabling arm following upstream suggestion is acceptable, but limiting to single arch due to precompiled components is not.

> I really don't know why it always runs out of time.

See linked PR.

**Comment by shizonic** [comment]
> but limiting to single arch due to precompiled components is not.

It's not because of that, it's because x64_86 is currently the only supported arch. See: https://github.com/denoland/deno/issues/7958#issuecomment-708003826

**Comment by ericonr** [comment]
You might need to package `GN` in void, I think (ew). https://github.com/denoland/rusty_v8/blob/master/build.rs#L282 I think the env var is just `GN`.

**Comment by shizonic** [comment]
@ericonr Thanks! What should the content of the env var `GN` be? The path to the gn binary?

**Comment by shizonic** [comment]
Waiting for merge of https://github.com/void-linux/void-packages/pull/25703

**Comment by shizonic** [comment]
`[ci skip]` is needed because rusty_v8 need more than travis' max time limit for jobs (50 mins) for building. But it build successfully on my local machine!

**Comment by Johnnynator** [comment]
```diff
diff --git a/srcpkgs/deno/patches/rusty_v8.patch b/srcpkgs/deno/patches/rusty_v8.patch
new file mode 100644
index 0000000000..a32cb05f82
--- /dev/null
+++ b/srcpkgs/deno/patches/rusty_v8.patch
@@ -0,0 +1,11 @@
+--- Cargo.lock	2020-10-10 12:30:55.000000000 +0200
++++ -	2020-10-22 17:12:50.248150438 +0200
+@@ -1920,8 +1920,6 @@
+ [[package]]
+ name = "rusty_v8"
+ version = "0.11.0"
+-source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "8c11463988ec37b3f8cb84e4c2fe8f63058c46e07348b6d1d27b114a2b981304"
+ dependencies = [
+  "bitflags",
+  "cargo_gn",
diff --git a/srcpkgs/deno/template b/srcpkgs/deno/template
index a790886bac..1d68e8c486 100644
--- a/srcpkgs/deno/template
+++ b/srcpkgs/deno/template
@@ -2,20 +2,34 @@
 pkgname=deno
 version=1.4.6
 revision=1
-archs="x86_64"
 build_helper="rust"
 hostmakedepends="cargo pkg-config curl unzip python ninja gn clang nodejs
- gobject-introspection"
+ lld llvm gobject-introspection"
 depends="libglib-devel"
 short_desc="Simple, modern and secure runtime for JavaScript and TypeScript"
 maintainer="shizonic <realtiaz@gmail.com>"
 license="MIT"
 homepage="https://deno.land"
-distfiles="https://github.com/denoland/deno/archive/v${version}.tar.gz"
-checksum=802bd22bcb00c1f2331f054b83820dfcdd7f79cc135bcd062d057f35fec9cba4
+distfiles="https://github.com/denoland/deno/archive/v${version}.tar.gz
+ https://void.johnnynator.dev/distfiles/rusty_v8-0.11.0.tar.gz"
+checksum="802bd22bcb00c1f2331f054b83820dfcdd7f79cc135bcd062d057f35fec9cba4
+ c4045d5a3e7f95f6faa323639a3289f7c543385c10a9015f73dfe97fc7d6ca7c"
+
+post_extract() {
+	mv ../rusty_v8-0.11.0 rusty-void
+	cat >>Cargo.toml <<-'_EOF'
+	
+	[patch.crates-io]
+	rusty_v8 = { path = 'rusty-void' }
+	_EOF
+}
 
 do_build() {
-	V8_FROM_SOURCE=1 cargo build --release --locked --target ${RUST_TARGET} -vv
+	CLANG_BASE_PATH=/usr \
+		NINJA=/usr/bin/ninja \
+		GN=/usr/bin/gn \
+		V8_FROM_SOURCE=1 \
+		cargo build --release --locked --target ${RUST_TARGET} -vv
 }
 
 do_install() {
diff --git a/srcpkgs/gn/template b/srcpkgs/gn/template
index ae54fb17f2..155a81ae0e 100644
--- a/srcpkgs/gn/template
+++ b/srcpkgs/gn/template
@@ -2,7 +2,7 @@
 pkgname=gn
 version=0.0.20201022
 revision=1
-_githash=f5f465b52e5df6d4de9fbe67cc79f1914ba4dbfc
+_githash=6f13aaac
 create_wrksrc=yes
 hostmakedepends="python3 ninja"
 short_desc="Meta-build system that generates build files for Ninja"
@@ -10,7 +10,7 @@ maintainer="shizonic <realtiaz@gmail.com>"
 license="BSD-3-Clause"
 homepage="https://gn.googlesource.com/gn/"
 distfiles="https://gn.googlesource.com/gn/+archive/${_githash}.tar.gz"
-checksum=@3ef35abf6ff8530e55cd145c7f9498d6e08169d5fea9d9ce4ece29ff4b7fb14a
+checksum=@80d9da7d58252fa49089386c943d8e3c493a4e314309dbf5e7b7585bfb484fa8
 
 post_extract() {
 	printf '#define LAST_COMMIT_POSITION "%s"\n' "$version" >src/gn/last_commit_position.h
```

I could make a proper from source build with this patch ontop. (The rusty_v8-0.11.0.tar.gz tarball is due to cargo beeing annoying when you want to patch, and it also doesn't do a shallow clone when pointed to a git repo)

**Comment by shizonic** [comment]
Gonna test it ASAP. Thank you!

**Comment by shizonic** [comment]
> The rusty_v8-0.11.0.tar.gz tarball is due to cargo beeing annoying when you want to patch, and it also doesn't do a shallow clone when pointed to a git repo

May you explain this further? How did you build this tarball? Is it a own pkg template?

**Comment by Johnnynator** [comment]
> > The rusty_v8-0.11.0.tar.gz tarball is due to cargo beeing annoying when you want to patch, and it also doesn't do a shallow clone when pointed to a git repo
> 
> May you explain this further? How did you build this tarball? Is it a own pkg template?

It is essentially this branch: https://github.com/Johnnynator/chromium_build/tree/void and created with https://pypi.org/project/git-archive-all/ . This is the only patch on that compared to the official 0.11.0 https://github.com/Johnnynator/chromium_build/commit/e33d3eae5f2b37e39c96b9dfb9d579fbf09d5f5b . I had to got the tarball route, to prevent cargo cloning 1+Gb of files.

**Comment by shizonic** [comment]
> > > The rusty_v8-0.11.0.tar.gz tarball is due to cargo beeing annoying when you want to patch, and it also doesn't do a shallow clone when pointed to a git repo
> > 
> > 
> > May you explain this further? How did you build this tarball? Is it a own pkg template?
> 
> It is essentially this branch: https://github.com/Johnnynator/chromium_build/tree/void and created with https://pypi.org/project/git-archive-all/ . This is the only patch on that compared to the official 0.11.0 [Johnnynator/chromium_build@e33d3ea](https://github.com/Johnnynator/chromium_build/commit/e33d3eae5f2b37e39c96b9dfb9d579fbf09d5f5b) . I had to got the tarball route, to prevent cargo cloning 1+Gb of files.

Thx for explanation. I see the use of the tarball-way. Creative indeed. 👍

**Comment by smorimoto** [comment]
Since we are using GitHub Actions now, we may be able to build without skipping.

**Comment by shizonic** [comment]
Thanks for informing. Really great news and I'm gonna update the template ASAP.

**Comment by smorimoto** [comment]
@shizonic Thanks for your efforts on this :)

**Comment by q66** [comment]
it's come to my attention that `gn` is unstable, so it probably doesn't make sense to have a separate package of it; i'd just fetch it as a second distfile in the `deno` template and build it as a part of it

**Comment by smorimoto** [comment]
The CIs should complete successfully, but the GitHub Actions runner is somehow hung and needs to be triggered again.

**Comment by smorimoto** [comment]
Oh disregard. It looks like it's still running.

**Comment by shizonic** [comment]
> Oh disregard. It looks like it's still running.

Yes, build process needs terribly long...

**Comment by shizonic** [comment]
builds now fine on x86_64. Other archs aren't supported yet...

**Comment by smorimoto** [comment]
Checks have passed.

**Comment by shizonic** [comment]
> Checks have passed.

Not at all: travis-ci still failes even if I don't understand why these checks are still present.

**Comment by smorimoto** [comment]
Well, I think we can ignore the result of Travis.

**Comment by shizonic** [comment]
> Well, I think we can ignore the result of Travis.

Agree but guess it still won't get merged because it's only for x86_64 arch... :(

**Comment by shizonic** [comment]
Is there even a chance to get this merged?

**Comment by q66** [comment]
only if it can do musl and/or other archs imo

**Comment by smorimoto** [comment]
​I think there is no reason why we cannot support all of them. (​In fact, I built and used Deno on aarch64 machines.)

**Comment by q66** [comment]
definitely not all of them, i just ran a `ppc64le` build out of curiosity and it fails on `ring` which seems to be a hard dependency and practically restricts the whole thing to `x86_64` and possibly `aarch64`

**Comment by shizonic** [comment]
Okay, to do so I need help. I only get it building on x86_64 and AFAIK officially it's only supported on this too.

**Comment by kawaiiamber** [comment]
Even if it's not officially supported, I'd like to see it built on `musl` libc, as well, being it's what I use teehee. As for how to test against other archs locally, `xbps-src` has an `-a` flag. You can `./xbps-src -a x86_64-musl pkg deno` if you wanted to test, for example.

I just read the conversation, and, uhh. I wonder why not just use `build_style=cargo`? Does it not work?

**Comment by Chocimier** [comment]
> You can ./xbps-src -a x86_64-musl pkg deno if you wanted to test, for example.

That's not what builders do. -a is for other processor. For other libc separate masterdir is better, see readme.

> I wonder why not just use build_style=cargo? Does it not work?

Try and tell us.

**Pull request closed without merging**